src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt

```diff
@@ -268,6 +272,11 @@ private fun SpacingContext.blockMustBeMultiLine(): Boolean {
     }
 }
 
+private fun SpacingContext.needsBlankLineBetweenItems(): Boolean {
+    // Allow to keep consecutive runs of `use`, `const` or `static` items without blank lines
+    return !(elementType1 == elementType2 && elementType1 in ONE_LINE_ITEMS)
```

Comment:
Hm, `ncPsi` didn't help me, but just pluggin `if either one is comment return false` seems to do the job. 

Still, I think the right approach here is to properly use comment binders, so, for example

```
mod foo;

//this comment is a part of the PSI node for bar, 
// so no need for special handling in formatter
mod bar        
```

I've tried to use simple gready left binder, but it breaks in the following case

```
mod foo; // this should be attached to foo, but is attached to `bar` with gready binder

mod bar; 
```